crates/lib3h_protocol/src/data_types.rs

```diff
@@ -6,6 +6,61 @@ use url::Url;
 /// (entry_address, aspect_address)
 pub type AspectKey = (Address, Address);
 
+/// Represents an opaque vector of bytes. Lib3h will
+/// store or transfer this data but will never inspect
+/// or interpret its contents
+#[derive(Clone, Eq, PartialEq, Deserialize, Serialize)]
+pub struct Opaque(#[serde(with = "base64")] Vec<u8>);
```

Comment:
should we shrinkwraprs, since we're pulling that in as a dependency for ghost_actor in any case? - https://github.com/holochain/lib3h/blob/ghost-actor/crates/ghost_actor/src/lib.rs#L25-L26

Reply:
That's a good idea. Will look into it for this PR or subsequent one depending on timing.